src/libecalc/common/utils/rates.py

```diff
@@ -531,6 +548,18 @@ def to_volumes(self) -> TimeSeriesVolumes:
 
 
 class TimeSeriesVolumes(TimeSeries[float]):
+    @validator("values", pre=True)
+    def check_length_timestep_values(cls, v: list, values, field: ModelField):
+        # Initially timesteps for volumes contains one more item than values
+        # After reindex number of timesteps equals number of values
+        # TODO: Ensure periodical volumes are handled in a consistent way. Why different after reindex?
+        if len(v) not in [len(values["timesteps"]), len(values["timesteps"]) - 1]:
```

Comment:
The periodical volumes initially have one more timestep than values. After reindexing this changes. This is why the check is looking for two values. This should be looked into. Note that periodical volumes are dependent on the actual timesteps (one more than number of periods), if we should continue to calculate cumulatives from periodical volumes in the same way as we do.

Reply:
Similar to StreamDay "problem", the timesteps should be in core, while volume should only be handled in the layer where we are reporting etc